['text':'!/usr/bin/env python','line_number':1,'multiline':False]['text':'','line_number':2,'multiline':False]['text':' Public Domain 2014-present MongoDB, Inc.','line_number':3,'multiline':False]['text':' Public Domain 2008-2014 WiredTiger, Inc.','line_number':4,'multiline':False]['text':'','line_number':5,'multiline':False]['text':' This is free and unencumbered software released into the public domain.','line_number':6,'multiline':False]['text':'','line_number':7,'multiline':False]['text':' Anyone is free to copy, modify, publish, use, compile, sell, or','line_number':8,'multiline':False]['text':' distribute this software, either in source code form or as a compiled','line_number':9,'multiline':False]['text':' binary, for any purpose, commercial or non-commercial, and by any','line_number':10,'multiline':False]['text':' means.','line_number':11,'multiline':False]['text':'','line_number':12,'multiline':False]['text':' In jurisdictions that recognize copyright laws, the author or authors','line_number':13,'multiline':False]['text':' of this software dedicate any and all copyright interest in the','line_number':14,'multiline':False]['text':' software to the public domain. We make this dedication for the benefit','line_number':15,'multiline':False]['text':' of the public at large and to the detriment of our heirs and','line_number':16,'multiline':False]['text':' successors. We intend this dedication to be an overt act of','line_number':17,'multiline':False]['text':' relinquishment in perpetuity of all present and future rights to this','line_number':18,'multiline':False]['text':' software under copyright law.','line_number':19,'multiline':False]['text':'','line_number':20,'multiline':False]['text':' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,','line_number':21,'multiline':False]['text':' EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF','line_number':22,'multiline':False]['text':' MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.','line_number':23,'multiline':False]['text':' IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR','line_number':24,'multiline':False]['text':' OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,','line_number':25,'multiline':False]['text':' ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR','line_number':26,'multiline':False]['text':' OTHER DEALINGS IN THE SOFTWARE.','line_number':27,'multiline':False]['text':' test_debug_mode05.py','line_number':31,'multiline':False]['text':'     As per WT-5046, the debug table logging settings prevent rollback to','line_number':32,'multiline':False]['text':'     stable in the presence of prepared transactions.','line_number':33,'multiline':False]['text':'','line_number':34,'multiline':False]['text':'     This test is to confirm the fix and prevent similar regressions.','line_number':35,'multiline':False]['text':' Try doing a normal prepared txn and then rollback to stable.','line_number':46,'multiline':False]['text':' The original bug happened when we had a txn that:','line_number':62,'multiline':False]['text':' 1. Was prepared.','line_number':63,'multiline':False]['text':' 2. Did not cause anything to be written to the log before committing.','line_number':64,'multiline':False]['text':' 3. Was the last txn before the rollback to stable call.','line_number':65,'multiline':False]['text':' Therefore, we're specifically not doing any operations here.','line_number':66,'multiline':False]['text':' The aforementioned bug resulted in a failure in rollback to stable.','line_number':76,'multiline':False]['text':' This is because we failed to clear out a txn id from our global state','line_number':77,'multiline':False]['text':' which caused us to think that we had a running txn.','line_number':78,'multiline':False]['text':' Verify that we can rollback to stable without issues.','line_number':79,'multiline':False]